### PR TITLE
Add aten mkldnn zero_ operator

### PR DESCRIPTION
**Pull request by XiaobingSuper**
### mkldnn backward ops list:
 - [ ] \(https://github.com/pytorch/pytorch/pull/20567) Add aten mkldnn conv2d backward operator :yellow_heart:
 - [ ] \(https://github.com/pytorch/pytorch/pull/20570) Add aten mkldnn backward ops: relu, linear and reshape :yellow_heart:
 - [ ] \(https://github.com/pytorch/pytorch/pull/20571) Add aten mkldnn backward ops: max_pool2d, avg_pool2d and adaptive_avg_poo2d :yellow_heart:
 - [ ] \(https://github.com/pytorch/pytorch/pull/20572) Add aten mkldnn batchnorm backward operator :yellow_heart:
 - [x] \(https://github.com/pytorch/pytorch/pull/20573) Add aten mkldnn zero_ operator💚 
 - [ ] \(https://github.com/pytorch/pytorch/pull/20575) Add mkldnn mul operator 💚

### CODE REVIEW

**Review by bddppq — COMMENTED** [review]
There are merge conflicts and failed CI jobs, could you rebase?

**Comment by XiaobingSuper** [comment]
@bddppq , rebased it, but the failed cases are not realsted my changes, could you help see them? thanks!

File: aten/src/ATen/native/mkldnn/MkldnnTensorMath.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+Tensor& mkldnn_zero_(Tensor& self) {
+  using Vec = vec256::Vec256<float>;
+
+  ideep::tensor& x = itensor_from_mkldnn(self);
+
+  auto n = x.get_nelems();
+  auto* x_ = static_cast<float *>(x.get_data_handle());
+  parallel_for(0, n, 2048, [x_](int64_t begin, int64_t end){
+    vec256::map(
```
Comment (bddppq) [inline aten/src/ATen/native/mkldnn/MkldnnTensorMath.cpp]: @cpuhrsch Is it safe to unconditionally using these vectorized utilities? Or do we need to guard it by some avx macros check?

**Review by bddppq — APPROVED** [review]

File: aten/src/ATen/native/mkldnn/MkldnnTensorMath.cpp (hunk 1 above)
Comment (cpuhrsch) [reply aten/src/ATen/native/mkldnn/MkldnnTensorMath.cpp]: Vec256 uses __AVX__ and __AVX2__ guards to compile with the compatibility version. So calling https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/cpu/vec256/functional.h#L99 without compiling it for avx2 is correct.

But, since this isn't compiled with avx2 for OSS, it might be slow for users. Why isn't this a kernel?

Comment (XiaobingSuper) [reply aten/src/ATen/native/mkldnn/MkldnnTensorMath.cpp]: @cpuhrsch , if written as a kernel, the input should not be a TensorIterator  as [add_kernel](https://github.com/pytorch/pytorch/blob/cbf572671d4c46fa7bd5cd0f8a01ca40f623d6c6/aten/src/ATen/native/cpu/BinaryOpsKernel.cpp#L16) , or written as [SoftMaxKernel](https://github.com/pytorch/pytorch/blob/cbf572671d4c46fa7bd5cd0f8a01ca40f623d6c6/aten/src/ATen/native/cpu/SoftMaxKernel.cpp#L25), please give me some advice, thanks!

Comment (cpuhrsch) [reply aten/src/ATen/native/mkldnn/MkldnnTensorMath.cpp]: cc @VitalyFedyunin

Comment (VitalyFedyunin) [reply aten/src/ATen/native/mkldnn/MkldnnTensorMath.cpp]: Inputs doesn't have to be TensorIterator. We prefer TensorIterator as input for kernels as we use same iterator for CPU and CUDA. In your case you can implement kernel that takes Tensor as input.

Comment (VitalyFedyunin) [reply aten/src/ATen/native/mkldnn/MkldnnTensorMath.cpp]: Using map here is also inefficient, please consider writing something like vec256::fill which will loadu vector, fill it and store.

Comment (XiaobingSuper) [reply aten/src/ATen/native/mkldnn/MkldnnTensorMath.cpp]: @VitalyFedyunin , thanks for your suggestion, I will change the code.

Comment (XiaobingSuper) [reply aten/src/ATen/native/mkldnn/MkldnnTensorMath.cpp]: @VitalyFedyunin , I have added a fill API in ideep, perharps direct process opaque tensor is not good in pytorch,  After talking with @jgong5,  add soomthing like vec256::fill is only useful mkldnn tensor, perhaps this way have not big benefit. Thanks!
